### PR TITLE
feat: allow to pass vault_url variable to indicate external Vault setup

### LINKED ISSUE #95: Allow to provide external Vault configuration
The latest jx version allows using an internal (using the Vault operator) or external Vault instance for secret storage. The Terraform module should allow configuring either of these two options.

See also https://github.com/jenkins-x/terraform-google-jx/issues/104

### PR DESCRIPTION
**Pull request by hferentschik**
- if vault_url is specified Vault module resources are not generated
- the resulting requirements file contains the Vault URL, but needs ti be edited with remaining external Vault configuration before running Boot

fixes #95

### CODE REVIEW

File: modules/vault/main.tf [hunk 1]
```diff
@@ -80,7 +88,7 @@ resource "aws_kms_key" "kms_vault_unseal" {
             "Effect": "Allow",
             "Principal": {
                 "AWS": [
-                    "${data.aws_iam_user.vault_user.arn}",
+                    "${data.aws_iam_user.vault_user[0].arn}",
```
Comment (ankitm123) [inline modules/vault/main.tf:91]: I know this is still a WIP, but probably better to check if `aws_iam_user.vault_user` is a non-empty tuple, before trying to access it, thoughts?

File: modules/vault/outputs.tf [hunk 2]
```diff
@@ -3,21 +3,22 @@
...
+    value = length(aws_kms_key.kms_vault_unseal) > 0 ? aws_kms_key.kms_vault_unseal[0].id : ""
+    
 }
 
 // ----------------------------------------------------------------------------
 // The created S3 Bucket ID
 // ----------------------------------------------------------------------------
 output "vault_unseal_bucket" {
-    value = aws_s3_bucket.vault-unseal-bucket.id
+    value = length(aws_s3_bucket.vault-unseal-bucket) > 0 ? aws_s3_bucket.vault-unseal-bucket[0].id : ""
```
Comment (ankitm123) [inline modules/vault/outputs.tf:14]: :+1:

File: modules/vault/main.tf (hunk 1 above)
Comment (hferentschik) [reply modules/vault/main.tf:91]: Maybe. For good measure. I was coming from that all resources are conditional on the `var.external_vault` condition. So assuming this resource (resource "aws_kms_key" "kms_vault_unseal") gets created all others are getting created as well and due to the dependencies between them, I know it exist.

Comment (hferentschik) [reply modules/vault/main.tf]: I added the check

File: spec/test_spec.sh [hunk 3]
```diff
@@ -8,84 +8,117 @@ Describe "AWS"
...
       When call aws dynamodb describe-table --table-name=$(terraform output vault_dynamodb_table)
       The output should not include ResourceNotFoundException
     End
+
     It "The KMS Key should be created"
       When call aws kms describe-key --key-id=$(terraform output vault_kms_unseal)
       The output should not include NotFoundException
     End
   End
+
```
Comment (ankitm123) [inline spec/test_spec.sh:109]: Unrelated to this PR, I will add terratests by the end of this week.

Comment (hferentschik) [reply spec/test_spec.sh:109]: hmm, do you think it is worth pulling Go into this? What do we gain? Let's open an issue and discuss.

Comment (ankitm123) [reply spec/test_spec.sh:109]: I created this issue sometime back: https://github.com/jenkins-x/terraform-aws-eks-jx/issues/30

**Pull request merged**